src/blocks/posts/index.php

```diff
@@ -0,0 +1,417 @@
+				esc_url( $post['thumbnailURL'] )
+			);
+
+			if ( 'horizontal' === $block_style && ( isset( $attributes['displayPostContent'] ) && ! $attributes['displayPostContent'] ) && ( isset( $attributes['columns'] ) && 2 >= $attributes['columns'] ) ) {
+
+				$align_self_class = 'self-center';
+			}
+		} else {
+			$align_self_class = 'flex-start';
+		}
+
+		$list_items_markup .= sprintf(
+			'<div class="wp-block-coblocks-posts__content flex flex-col %s w-full">',
+			esc_attr( $align_self_class )
+		);
+
+		if ( isset( $attributes['displayPostDate'] ) && $attributes['displayPostDate'] && 'stacked' === $block_style ) {
+
+			$list_items_markup .= sprintf(
+				'<time datetime="%1$s" class="wp-block-coblocks-posts__date mb-1">%2$s</time>',
+				esc_url( $post['date'] ),
+				esc_html( $post['dateReadable'] )
+			);
+
+		}
+
+		$title = $post['title'];
+
+		if ( ! $title ) {
+
+			$title = _x( '(no title)', 'placeholder when a post has no title', 'coblocks' );
+
+		}
+
+		$list_items_markup .= sprintf(
+			'<a href="%1$s" alt="%2$s">%2$s</a>',
+			$post['postLink'],
+			$title
+		);
+
+		if ( isset( $attributes['displayPostDate'] ) && $attributes['displayPostDate'] && 'horizontal' === $block_style ) {
+
+			$list_items_markup .= sprintf(
+				'<time datetime="%1$s" class="wp-block-coblocks-posts__date mt-2">%2$s</time>',
+				esc_url( $post['date'] ),
+				esc_html( $post['dateReadable'] )
+			);
+
+		}
+
+		if ( isset( $attributes['displayPostContent'] ) && $attributes['displayPostContent'] ) {
+
+			$post_excerpt    = $post['postExcerpt'];
+			$trimmed_excerpt = esc_html( wp_trim_words( $post_excerpt, $attributes['excerptLength'], ' &hellip; ' ) );
+
+			$list_items_markup .= sprintf(
+				'<div class="wp-block-coblocks-posts__post-excerpt mt-1">%1$s</div>',
+				esc_html( $trimmed_excerpt )
+			);
+
+		}
+
+		if ( isset( $attributes['displayPostLink'] ) && $attributes['displayPostLink'] ) {
+
+			$list_items_markup .= sprintf(
+				'<div class="wp-block-coblocks-posts__more-link"><a href="%1$s">%2$s</a></div>',
+				esc_url( $post['postLink'] ),
+				esc_html( $attributes['postLink'] )
+			);
+
+		}
+
+		$list_items_markup .= '</li>';
+
+	}
+
+	$block_content .= $list_items_markup;
+	$block_content .= '</ul>';
+
+	return $block_content;
+
+}
+
+/**
+ * Returns the posts for an internal posts.
+ *
+ * @param array $posts Current posts.
+ *
+ * @return array Returns posts.
+ */
+function coblocks_get_post_info( $posts ) {
+
+	$formatted_posts = [];
+
+	foreach ( $posts as $post ) {
+
+		$formatted_post = null;
+
+		$formatted_post['thumbnailURL'] = get_the_post_thumbnail_url( $post );
+		$formatted_post['date']         = esc_attr( get_the_date( 'c', $post ) );
+		$formatted_post['dateReadable'] = esc_html( get_the_date( '', $post ) );
+		$formatted_post['title']        = get_the_title( $post );
+		$formatted_post['postLink']     = esc_url( get_permalink( $post ) );
+
+		$post_excerpt = $post->post_excerpt;
+
+		if ( ! ( $post_excerpt ) ) {
```

Comment:
We do not need to check if the post excerpt exists. The WordPress function [`get_the_excerpt`](https://developer.wordpress.org/reference/functions/get_the_excerpt/) will automatically generate an excerpt based on the content that exists.